plugin/src/main/java/org/opensearch/ml/processor/ModelExecutor.java

```diff
@@ -0,0 +1,176 @@
+import org.opensearch.ml.common.input.MLInput;
+import org.opensearch.ml.common.transport.prediction.MLPredictionTaskRequest;
+
+import com.jayway.jsonpath.Configuration;
+import com.jayway.jsonpath.JsonPath;
+import com.jayway.jsonpath.Option;
+
+/**
+ * General ModelExecutor interface.
+ */
+public interface ModelExecutor {
+
+    Configuration suppressExceptionConfiguration = Configuration
+        .builder()
+        .options(Option.SUPPRESS_EXCEPTIONS, Option.DEFAULT_PATH_LEAF_TO_NULL)
+        .build();
+
+    default <T> ActionRequest getRemoteModelInferenceRequest(Map<String, String> parameters, String modelId) {
+
+        RemoteInferenceInputDataSet inputDataSet = RemoteInferenceInputDataSet.builder().parameters(parameters).build();
+        if (inputDataSet.getParameters() == null) {
+            throw new IllegalArgumentException("wrong input. The model input cannot be empty.");
+        }
+        MLInput mlInput = MLInput.builder().algorithm(FunctionName.REMOTE).inputDataset(inputDataSet).build();
+
+        ActionRequest request = new MLPredictionTaskRequest(modelId, mlInput, null);
+
+        return request;
+
+    }
+
+    /**
+     * filter model outputs by field name,
+     * supported using dot path look up for objects
+     * when a field name not provided, default to get all prediction outputs
+     */
+    default Object getModelOutputField(Map<String, ?> modelTensorOutputMap, String fieldName, boolean ignoreMissing) throws IOException {
+
+        if (fieldName == null) {
+            return modelTensorOutputMap;
+        } else if (modelTensorOutputMap.containsKey(fieldName) && modelTensorOutputMap != null) {
+            Object filteredOutput = modelTensorOutputMap.get(fieldName);
+            return filteredOutput;
+        } else {
+            try {
+                // avoid assigning prediction values in all positions of array, remove the last [*]
+                String jsonPath = findDotPathForNestedObject(modelTensorOutputMap, fieldName).replaceAll("\\[\\*\\]$", "");
+                Object filteredOutput = JsonPath.read(modelTensorOutputMap, jsonPath);
+                return filteredOutput;
+            } catch (Exception e) {
+                if (ignoreMissing) {
+                    return modelTensorOutputMap;
+                } else {
+                    throw new IOException("model inference output can not find field name: " + fieldName, e);
+                }
+            }
+        }
+    }
+
+    /**
+     * cast object to JsonString
+     */
+
+    default String toString(Object originalFieldValue) {
+        String originalFieldValueAsString = gson.toJson(originalFieldValue);
+        return originalFieldValueAsString;
+    }
+
+    /**
+     * find json path in nested object based on dot path without explicit array locations
+     * return a json path
+     * for example foo.bar.quz to be  foo[*].bar.quiz[*]
+     */
+    default String findDotPathForNestedObject(Object json, String dotPath) {
```

Comment:
This method hard limit how to parse result, seems not flexible. How about just let user use the standard json path ?

Reply:
removed this method and accept json path only, please check this IT with the json path notation for nested object 

https://github.com/opensearch-project/ml-commons/blob/b9dfc86ad0d656530d1f112d11704828691bd335/plugin/src/test/java/org/opensearch/ml/rest/RestMLInferenceIngestProcessorIT.java#L144